policy-controller/k8s/status/src/index.rs

```diff
@@ -325,6 +320,39 @@ impl kubert::index::IndexNamespacedResource<k8s::policy::Server> for Index {
     // to handle resets specially.
 }
 
+impl kubert::index::IndexNamespacedResource<k8s::Service> for Index {
+    fn apply(&mut self, resource: k8s::Service) {
+        let namespace = resource.namespace().expect("Service must have a namespace");
+        let name = resource.name_unchecked();
+        let id = ResourceId::new(namespace, name);
+
+        self.services.insert(id);
```

Comment:
Should we perhaps hold more information in the cache aside from the namespace and name? For example, routes [should not attach to headless services](https://gateway-api.sigs.k8s.io/geps/gep-1426/#allowed-service-types). Similarly, services that have an `ExternalName` type are not meant to be attached to either due to security concerns.

Do we want to consider this when setting the status?

Reply:
Ah, I wasn't aware of these restrictions.  However, they make perfect sense.  Let's tackle in a follow up.